tools/export/codeblocks/cbp.tmpl

```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
+<CodeBlocks_project_file>
+	<FileVersion major="1" minor="6" />
+	<Project>
+		<Option title="{{project_name}}" />
+		<Option pch_mode="2" />
+		<Option compiler="arm-elf-gcc" />
+		<Build>
+			<Target title="Debug">
+				<Option output="bin/Debug/{{project_name}}.elf" prefix_auto="1" extension_auto="0" />
+				<Option object_output="obj/Debug/" />
+				<Option type="1" />
+				<Option compiler="arm-elf-gcc" />
+				<Option use_console_runner="0" />
+				<Compiler>
+					{%  for f in debug_flags -%}
+					<Add option="{{f}}" />
+					{%  endfor -%}
+				</Compiler>
+				<Linker>
+					<Add option='-Wl,-Map,&quot;bin/Debug/{{project_name}}.map&quot;' />
+				</Linker>
+				{% if ncs36510addfib -%}
+				<ExtraCommands>
+					<Add after="arm-none-eabi-objcopy -O ihex -R .fib -R .trim bin/Debug/{{project_name}}.elf bin/Debug/{{project_name}}.hex" />
+					<Add after="mbed-os/tools/export/codeblocks/ncs36510addfib.py bin/Debug/{{project_name}}.hex bin/Debug/{{project_name}}.fib" />
```

Comment:
This is not allowed: after export on the website, a project will not contain the tools.

Reply:
Where would be the proper place for such a tool?

Reply:
This is where things get tricky. You could export that python script so that it is in the root of the project, then you can rely on it's existence. You can rely on anything in resources, and anything you explicitly create, so if you explicitly create the file, you're good to go. I, and possibly future users, would appreciate an addition to [the third party notes section](https://github.com/ARMmbed/Handbook/blob/new_engine/docs/tools/toolchains/export_to_third_party.md#third-party-tool-notes) explaining that you need Python to compile with Code::Blocks and NCS36510.